Add test for #32108 (error with groupby on series with period index)

daxid opened the pull request:
- [ ] closes #32108

dbeniamine committed:
Add test for #32108

pep8speaks commented:
Hello @daxid! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/tests/groupby/test_groupby.py`](https://github.com/pandas-dev/pandas/blob/74e65a8f9372b893a3d228973fdd9472de971ff3/pandas/tests/groupby/test_groupby.py):

> [Line 2062:1](https://github.com/pandas-dev/pandas/blob/74e65a8f9372b893a3d228973fdd9472de971ff3/pandas/tests/groupby/test_groupby.py#L2062): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 2063:12](https://github.com/pandas-dev/pandas/blob/74e65a8f9372b893a3d228973fdd9472de971ff3/pandas/tests/groupby/test_groupby.py#L2063): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 2069:33](https://github.com/pandas-dev/pandas/blob/74e65a8f9372b893a3d228973fdd9472de971ff3/pandas/tests/groupby/test_groupby.py#L2069): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 2069:57](https://github.com/pandas-dev/pandas/blob/74e65a8f9372b893a3d228973fdd9472de971ff3/pandas/tests/groupby/test_groupby.py#L2069): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 2069:65](https://github.com/pandas-dev/pandas/blob/74e65a8f9372b893a3d228973fdd9472de971ff3/pandas/tests/groupby/test_groupby.py#L2069): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator

daxid committed:
Typographic improvements in test_groupby.py

    Line 2062:1: E302 expected 2 blank lines, found 1
    Line 2063:12: E225 missing whitespace around operator
    Line 2069:33: E231 missing whitespace after ','
    Line 2069:57: E231 missing whitespace after ','
    Line 2069:65: E226 missing whitespace around arithmetic operator

daxid committed:
Yet again improve typo in test_groupby.py

##[error]pandas/tests/groupby/test_groupby.py:2062:         <- trailing whitespaces found
##[error]./pandas/tests/groupby/test_groupby.py:15:1:F811:redefinition of unused 'Series' from line 11
##[error]ERROR: /home/runner/work/pandas/pandas/pandas/tests/groupby/test_groupby.py Imports are incorrectly sorted.

MarcoGorelli commented on pandas/tests/groupby/test_groupby.py:
```diff
@@ -2057,3 +2057,15 @@ def test_groups_repr_truncates(max_seq_items, expected):
         assert result == expected
+
+
+def test_groupby_period_index():
```
Add the issue number here:

```
def test_groupby_period_index():
    # GH 32108
```

MarcoGorelli commented on pandas/tests/groupby/test_groupby.py:
```diff
@@ -2057,3 +2057,15 @@ def test_groups_repr_truncates(max_seq_items, expected):
+
+def test_groupby_period_index():
+    periods = 2
+    index = period_range(start='2018-01', periods=periods, freq='M')
```
Looks like you haven't run `black`, and perhaps not `flake8` either. See [code standards](https://pandas.pydata.org/docs/development/contributing.html#code-standards)

MarcoGorelli reviewed (changes requested):
Thanks @daxid !

dbeniamine committed:
Fix linter issues

dbeniamine committed:
Black formating

dbeniamine commented:
I've done the corrections, all tests are passing now, sorry for the delay

jreback commented on pandas/tests/groupby/test_groupby.py:
```diff
@@ -2057,3 +2057,16 @@ def test_groups_repr_truncates(max_seq_items, expected):
         result = df.groupby(np.array(df.a)).groups.__repr__()
         assert result == expected
+
+
```
can you move this under  test_grouper_index_types (same file). ping on green.

dbeniamine replied:
The function `test_grouper_index_types` is in the file `pandas/tests/groupby/test_grouping.py` our test is in `pandas/tests/groupby/test_groupby.py`, do you want us to move our test to `test_grouping.py` ?

jreback replied:
yes pls move this

dbeniamine replied:
done

dbeniamine committed:
Move test to test_grouping.py

MarcoGorelli commented on pandas/tests/groupby/test_grouping.py:
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
+        # GH 32108
+        periods = 2
+        index = pd.period_range(start="2018-01", periods=periods, freq="M")
+        periodSerie = pd.Series(range(periods), index=index)
```
naming this `periodSerie` doesn't conform to PEP8 - sorry for not bringing this up sooner, I thought flake8 would've caught it. seems there's an open issue to add extra plugins though (#20588)

Function and Variable Names
Function names should be lowercase, with words separated by underscores as necessary to improve readability.

> Variable names follow the same convention as function names.
> 
> mixedCase is allowed only in contexts where that's already the prevailing style (e.g. threading.py), to retain backwards compatibility.

https://www.python.org/dev/peps/pep-0008/

dbeniamine replied:
Done, I hope it's all good now :)

dbeniamine committed:
Conform to PEP8

simonjayhawkins commented:
@daxid thanks for working on this. seems to have been lost in the shuffle. can you merge master to trigger ci.

dbeniamine committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas

dbeniamine commented:
> @daxid thanks for working on this. seems to have been lost in the shuffle. can you merge master to trigger ci.

done

jreback commented:
can you merge master

dbeniamine committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas

dbeniamine commented:
> can you merge master

done

jreback commented on pandas/tests/groupby/test_grouping.py:
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
+        period_serie.index.name = "Month"
+        result = period_serie.groupby(period_serie.index.month).sum()
+
+        expected = pd.Series(range(0, periods), index=range(1, periods + 1))
```
you can construct the `Index(...., name=...)`

jreback commented on pandas/tests/groupby/test_grouping.py:
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
+    def test_grouper_period_index(self):
+        # GH 32108
+        periods = 2
+        index = pd.period_range(start="2018-01", periods=periods, freq="M")
```
pass `name=`

jreback commented on pandas/tests/groupby/test_grouping.py:
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
+        # GH 32108
+        periods = 2
+        index = pd.period_range(start="2018-01", periods=periods, freq="M")
+        period_serie = pd.Series(range(periods), index=index)
```
just name this period_series

jreback replied:
right can you construct the initial index like

```
> index = pd.period_range(start="2018-01", periods=periods, freq="M")
index = pd.period_range(start="2018-01", periods=periods, freq="M", name='foo')
```

dbeniamine replied:
The line on the current version is already:

```python
index = pd.period_range(
            start="2018-01", periods=periods, freq="M", name="Month"
        )
```

jreback commented on pandas/tests/groupby/test_grouping.py:
```diff
@@ -149,6 +149,18 @@ def test_grouper_index_types(self):
             df.groupby(list("abcde")).apply(lambda x: x)
 
+    def test_grouper_period_index(self):
+        # GH 32108
```
move this to pandas/tests/groupby/test_timegrouper.py

dbeniamine committed:
New modifications for PR

dbeniamine commented:
Done once again

simonjayhawkins commented on pandas/tests/groupby/test_timegrouper.py:
```diff
@@ -769,3 +769,18 @@ def test_scalar_call_versus_list_call(self):
+
+        expected = pd.Series(
+            range(0, periods),
+            index=Index(range(1, periods + 1), name=period_serie.index.name),
```
maybe have name as index.name or just be explicit and use name="Month"

simonjayhawkins reviewed (commented):
Thanks @daxid generally lgtm can you address https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823

dbeniamine committed:
Change index name

dbeniamine commented:
Is that version good for you ?

simonjayhawkins commented:
it appears that https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823 has not been addressed period_serie -> period_series otherwise lgtm. Thanks @daxid

dbeniamine commented:
> it appears that [#33105 (comment)](https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823) has not been addressed period_serie -> period_series otherwise lgtm. Thanks @daxid

I am not sure to understand you request, do you want us to add a `name=` parameter  in the `period_serie` construction or do you want us to change the name `period_serie` to something else, maybe `month_serie` ?

MarcoGorelli commented:
> > it appears that [#33105 (comment)](https://github.com/pandas-dev/pandas/pull/33105#discussion_r440134823) has not been addressed period_serie -> period_series otherwise lgtm. Thanks @daxid
> 
> I am not sure to understand you request, do you want us to add a `name=` parameter in the `period_serie` construction or do you want us to change the name `period_serie` to something else, maybe `month_serie` ?

Pretty sure the request is to rename `period_serie` to `period_series`

dbeniamine committed:
Rename period_series

dbeniamine commented:
> Pretty sure the request is to rename `period_serie` to `period_series`

Thanks did now saw the `s` so I wasn't understing this request, should be good now

jreback commented:
thanks @daxid
